VideoLocker/src/main/java/org/edx/mobile/loader/CoursesAsyncLoader.java

```diff
@@ -52,6 +45,18 @@ public CoursesAsyncLoader(Context context) {
         return result;
     }
 
+    private List<EnrolledCoursesResponse> getEnrolledCourses() {
+        List<EnrolledCoursesResponse> ret = null;
+
+        try {
+            ret = environment.getServiceManager().getEnrolledCourses(false);
```

Comment:
Well, there seems to be a back and forth on this, because it seems like we
WANT to use retrofit through ServiceManager (something to do with OAuth).

Retrofit may have their own cache, but we need to make sure that our code
has granular access over it. This is why I am proposing having the
ServiceManager imply contain its own cache.

On Wed, May 18, 2016 at 1:02 PM, Akiva Leffert <notifications@github.com>
wrote:

> In VideoLocker/src/main/java/org/edx/mobile/loader/CoursesAsyncLoader.java
> <https://github.com/edx/edx-app-android/pull/707#discussion_r63742169>:
>
> > @@ -52,6 +45,18 @@ public CoursesAsyncLoader(Context context) {
> >          return result;
> >      }
> >
> > +    private List<EnrolledCoursesResponse> getEnrolledCourses() {
> > +        List<EnrolledCoursesResponse> ret = null;
> > +
> > +        try {
> > +            ret = environment.getServiceManager().getEnrolledCourses(false);
>
> IMO, the way to do this is to not bother with retrofit. That way you can
> put the caching logic in one place instead of having to wrap every single
> method. Had some discussions with @brianguertin
> <https://github.com/brianguertin> and @BenjiLee
> <https://github.com/BenjiLee> about this recently.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-app-android/pull/707/files/0c45f11a71cef83eb283660b17c3042aa536934f#r63742169>
>



-- 
Regards,

Marc DiNino
Android Engineer
EDX